apps/wallet/src/ui/app/components/receipt-card/index.tsx

```diff
@@ -21,71 +20,93 @@ type TxResponseProps = {
     tranferType?: 'nft' | 'coin' | null;
 };
 
-function ReceiptCard({ tranferType, txDigest }: TxResponseProps) {
-    const TxIcon = txDigest.isSender ? SuiIcons.ArrowLeft : SuiIcons.Buy;
-    const iconClassName = txDigest.isSender
-        ? cl(st.arrowActionIcon, st.angledArrow)
-        : cl(st.arrowActionIcon, st.buyIcon);
+const TRUNCATE_MAX_LENGTH = 8;
+const TRUNCATE_PREFIX_LENGTH = 4;
 
-    const intl = useIntl();
+// Truncate text after one line (~ 35 characters)
+const TRUNCATE_MAX_CHAR = 40;
 
-    const imgUrl = txDigest?.url
-        ? txDigest?.url.replace(/^ipfs:\/\//, 'https://ipfs.io/ipfs/')
-        : false;
+function ReceiptCard({ txDigest }: TxResponseProps) {
+    const toAddrStr = useMiddleEllipsis(
+        txDigest.to || '',
+        TRUNCATE_MAX_LENGTH,
+        TRUNCATE_PREFIX_LENGTH
+    );
+    const fromAddrStr = useMiddleEllipsis(
+        txDigest.from || '',
+        TRUNCATE_MAX_LENGTH,
+        TRUNCATE_PREFIX_LENGTH
+    );
 
-    const date = txDigest?.timestampMs
-        ? formatDate(txDigest.timestampMs, ['month', 'day', 'year'])
-        : false;
+    const truncatedNftName = useMiddleEllipsis(
+        txDigest?.name || '',
+        TRUNCATE_MAX_CHAR,
+        TRUNCATE_MAX_CHAR - 1
+    );
+    const truncatedNftDiscription = useMiddleEllipsis(
+        txDigest?.description || '',
+        TRUNCATE_MAX_CHAR,
+        TRUNCATE_MAX_CHAR - 1
+    );
 
-    const transfersTxt = {
-        nft: {
-            header: 'Successfully Sent!',
+    const transferType =
+        txDigest.kind === 'Call'
+            ? 'Call'
+            : txDigest.isSender
+            ? 'Sent'
+            : 'Received';
+
+    const transferMeta = {
+        Call: {
+            txName: 'Minted',
```

Comment:
Can we just call this `Call`?

Reply:
The design has it as 'Minted'

Reply:
@Jordan-Mysten did you have a plan for how we could get the human-readable "Minted" verbiage in future if we use "Call" today?

Reply:
@mystie711 Yeah we're working on capturing some requirements and developing a solution for this space